Source/CarthageKit/Project.swift

```diff
@@ -534,22 +556,39 @@ private func cacheDownloadedBinary(downloadURL: NSURL, toURL cachedURL: NSURL) -
 		}
 }
 
-/// Sends the URL to each framework bundle found in the given directory.
-private func frameworksInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+extension NSURL {
+	// The type identifier of the receiver, or an error if it was unable to be
+	// determined.
+	internal func typeIdentifier() -> Result<String, NSError> {
+		return try({ (error: NSErrorPointer) -> String? in
```

Comment:
We can omit the parentheses and use a trailing closure as: `try { error in ... }`